plugins/chat/lib/chat_channel_fetcher.rb

```diff
@@ -62,7 +62,7 @@ def self.generate_allowed_channel_ids_sql(guardian, exclude_dm_channels: false)
   end
 
   def self.secured_public_channel_slug_lookup(guardian, slugs)
-    allowed_channel_ids = generate_allowed_channel_ids_sql(guardian, exclude_dm_channels: true)
+    allowed_channel_ids = generate_allowed_channel_ids_sql(guardian)
```

Comment:
I am not sure why you removed this one? This query doesn't need the DM channels.